pkg/controller/garbagecollector/dump_status.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package garbagecollector
+
+import (
+	"encoding/json"
+	"net/http"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// errorResult represents an error and its metadata that are to be marshalled into JSON.
+type errorResult struct {
```

Comment:
imo, this should be just `status` and below struct should be:
```golang
type statusResult struct {
    Items []status `json:"items"`
}
```

Reply:
currently the consensus is to make it simple and not part of the API. And revisit this later: https://github.com/kubernetes/kubernetes/pull/105705#discussion_r738301311